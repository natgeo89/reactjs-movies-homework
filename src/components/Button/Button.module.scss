.btn {
  min-width: 50px;
  padding: 10px;

  color: white;
  font-size: 2rem;

  background-color: transparent;
  border: 2px solid white;
  border-radius: 5px;
  cursor: pointer;
}

.primary {
  background-color: #09ade3;
  border: none;
}