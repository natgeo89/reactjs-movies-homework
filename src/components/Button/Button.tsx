import React from 'react';
import classNames from 'classnames';
import styles from './Button.module.scss';

interface ButtonProps {
  primary?: boolean;
  onClick?: () => void;
}

const Button: React.FC<ButtonProps> = ({ children, onClick, primary }) => {
  const classes = classNames(
    styles.btn,
    { [styles.primary]: primary },
  );
  return (
    <button className={classes} type="button" onClick={onClick}>{children}</button>
  );
};
export default Button;
