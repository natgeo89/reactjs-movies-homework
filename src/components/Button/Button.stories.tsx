import React from "react";

import Button from "./Button";

import { Story, Meta } from "@storybook/react";

export default {
  component: Button,
  title: "Components/Button",
} as Meta;

const Template: Story = (args) => <Button {...args} />;

export const Primary = Template.bind({});

export const Simple = Template.bind({});

Simple.args = {
  primary: false,
  children: "testtext",
};

Primary.args = {
  ...Simple.args,
  primary: true,
};

export const WithoutText = Template.bind({});

WithoutText.args = {
  ...Simple.args,
  children: "",
};
