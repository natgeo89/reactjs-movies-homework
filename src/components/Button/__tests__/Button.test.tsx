import React from "react";
import { render } from "@testing-library/react";
import Button from "../Button";

describe("Button", () => {
  test("snapshot", () => {
    const { asFragment } = render(<Button>Test</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
