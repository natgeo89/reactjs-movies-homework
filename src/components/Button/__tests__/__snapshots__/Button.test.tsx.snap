// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn"
    type="button"
  >
    Test
  </button>
</DocumentFragment>
`;
